Source/ASCollectionNode.mm

```diff
@@ -132,7 +147,7 @@ - (instancetype)initWithFrame:(CGRect)frame collectionViewLayout:(UICollectionVi
   ASDisplayNodeViewBlock collectionViewBlock = ^UIView *{
     // Variable will be unused if event logging is off.
     __unused __typeof__(self) strongSelf = weakSelf;
-    return [[ASCollectionView alloc] _initWithFrame:frame collectionViewLayout:layout layoutFacilitator:layoutFacilitator eventLog:ASDisplayNodeGetEventLog(strongSelf)];
+    return [[[strongSelf collectionViewClass] alloc] _initWithFrame:frame collectionViewLayout:layout layoutFacilitator:layoutFacilitator eventLog:ASDisplayNodeGetEventLog(strongSelf)];
```

Comment:
The use case is legitimate to me. 

Questions: Does the class need to be accessed later on? If yes, would checking `isKindOfClass:` on `collectionNode.view.class` be sufficient? Instead of having it as a property, can it be passed to an initializer as a parameter? Something like this: `initWithFrame:collectionViewClass:collectionViewLayout: layoutFacilitator:`.